sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,173 @@
+        ]
+
+        if self.get_option('add-namespaces'):
+            for nsp in self.get_option('add-namespaces').split(':'):
+                collect_regexes.append('^%s$' % nsp)
+
+        return collect_regexes
+
+    def _reduce_namespace_list(self, nsps):
+        """Reduce the namespace listing returned to just the ones we want to
+        collect from. By default, as requested by OCP support personnel, this
+        must include all 'openshift' prefixed namespaces
+
+            :param nsps list:            Namespace names from oc output
+        """
+
+        def _match_namespace(namespace):
+            """Match a particular namespace for inclusion (or not) in the
+            collection phases
+
+                :param namespace str:   The name of a namespace
+            """
+
+            for regex in self.collect_regexes:
+                if re.match(regex, namespace):
+                    return True
+            return False
+
+        self.collect_regexes = self._setup_namespace_regexes()
+
+        return list(set([n for n in nsps if _match_namespace(n)]))
+    
+    def _get_namespaces(self):
+        res = self.exec_cmd('microshift get namespaces -o custom-columns=NAME:.metadata.name --no-headers')
+        if res['status'] == 0:
+            return self._reduce_namespace_list(res['output'].split('\n'))
+        return []
+
+    def _collect_cluster_resources(self):
+        """Collect cluster-level (non-namespaced) resources
+        """
+        global_resources = [
+            'apiservices',
+            'certificatesigningrequests',
+            'clusterrolebindings',
+            'clusterroles',
+            'componentstatuses',
+            'csidrivers',
+            'csinodes',
+            'customresourcedefinitions',
+            'flowschemas',
+            'ingressclasses',
+            'logicalvolumes',
+            'mutatingwebhookconfigurations',
+            'nodes',
+            'persistentvolumes',
+            'priorityclasses',
+            'prioritylevelconfigurations',
+            'rangeallocations',
+            'runtimeclasses',
+            'securitycontextconstraints',
+            'selfsubjectaccessreviews',
+            'selfsubjectrulesreviews',
+            'storageclasses',
+            'subjectaccessreviews',
+            'tokenreviews',
+            'validatingwebhookconfigurations',
+            'volumeattachments'
+        ]
+
+        _filtered_resources = []
+
+        # Filter out any resources that might not exist.
+        for resource in global_resources:
+            res = self.exec_cmd("microshift get %s" % resource)
+            if res['status'] == 0:
+                _filtered_resources.append(resource)
+
+        _subdir = self.get_cmd_output_path(make=False)
+        self.exec_cmd(
+            'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
+                self.get_option('kubeconfig'),
+                _subdir,
+                ",".join(_filtered_resources)))
```

Comment:
You are not interested in the command output, but in the file it automatically generates, did I get it right?